romdisk/uart.asm

```diff
@@ -150,56 +176,127 @@ uartsnd_main:
         ;       A - 0 on success
         PUBLIC uartrcv_main
 uartrcv_main:
-        dec bc  ; Do not count the command itself
-        ld a, b
-        or c
-        ret z
+        ; (B)C must be 3, we want exactly three parameters (command itself, size, file)
+        ; B is 0 for sure
+        ld a, c
+        cp 3
+        jp nz, _uartrcv_usage
+        ; Retrieve the size (string) in DE destination file in HL
+        inc hl
+        inc hl
+        ld e, (hl)
         inc hl
+        ld d, (hl)
         inc hl
         ld a, (hl)
         inc hl
         ld h, (hl)
         ld l, a
+        ex de, hl  ;
+        ; Size in HL, destination in DE.
+
         ; Parse the length parameter
         call parse_int
         ; Check if A is 0 or not
         or a
-        ; At the moment, simply return, this command is just for the sake of demonstration
-        ret nz
-        ; Save the buffer size
+        jp nz, _uartrcv_usage
+
+        ; Save the buffer size on the stack
         push hl
+
+        ; Open the file, it is useless to receive data from the UART if the file cannot be created
+        ld h, O_WRONLY | O_CREAT | O_TRUNC
+        ; Filename in BC
+        ld b, d
+        ld c, e
+        OPEN()
+        or a
+        jp m, uart_file_error
+
+        ; File is opened, keep the descriptor on the stack and open the UART driver
+        ; Open the UART driver
+        ; BC - Path to file
+        ; H - Flags
+        push af
         call uart_open
-        ; Save the size in BC
+        jp m, uart_error_pop
+
+        ; Read from the UART driver
+        ld de, 0x8000   ; Buffer
+        ; Size is on the stack, below the file descriptor
+        pop hl
         pop bc
-        jp m, open_error
-        ; Read from the UART
-        ; DE - Buffer
-        ; BC - Size
-        ; H  - Descriptor
-        ld de, 0x8000
+        push hl
+        ; UART opened dev in H
         ld h, a
         READ()
-        ; TODO: Show what was actually received
-        ; Close the driver
+        ; TODO: Check BC to see if we read enough bytes
```

Comment:
Same issue, a file that is bigger than 16KB will see the remaining bytes ignored. This routine should loop until there are no remaining bytes to receive

Reply:
#wouldn't be solved